project/driver_controls/src/driver_io.c

```diff
@@ -0,0 +1,106 @@
+#include <stddef.h>
+
+#include "gpio_it.h"
+#include "driver_io.h"
+#include "driver_callback.h"
+
+// Arbitrary thresholds for gas pedal
+#define COAST_THRESHOLD 1000
+#define DRIVE_THRESHOLD 3000
+
+typedef struct DriverIOSettings {
```

Comment:
Why not just a few common GPIO configs? It seems like almost all of your IOs use the exactly same settings.